### DISCUSSION TITLE
Fresh Ubuntu 18.04 machine: /var/lib/teleport: permission denied

### DISCUSSION THREAD

**Discussion by vincentracine** [gravitational/teleport#13873] [Q&A] [answered]
I'm trying to set up a fresh server to run the cluster. I launched an Ubuntu 18.04 machine in AWS and tried following the docs https://goteleport.com/docs/getting-started/linux-server/. I'm run into the following error. Before I start opening the permissions for the `/var/lib/teleport` directory, I wanted to post here as I feel like I've followed the instructions and therefore other users might run into this issue.

```
ubuntu@ip-10-0-1-49:~$ sudo apt-get install teleport
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  teleport
0 upgraded, 1 newly installed, 0 to remove and 3 not upgraded.
Need to get 94.4 MB of archives.
After this operation, 331 MB of additional disk space will be used.
Get:1 https://deb.releases.teleport.dev stable/main amd64 teleport amd64 9.3.7 [94.4 MB]
Fetched 94.4 MB in 3s (31.1 MB/s)    
Selecting previously unselected package teleport.
(Reading database ... 58033 files and directories currently installed.)
Preparing to unpack .../teleport_9.3.7_amd64.deb ...
Unpacking teleport (9.3.7) ...
Setting up teleport (9.3.7) ...
ubuntu@ip-10-0-1-49:~$ teleport configure --acme --acme-email=vincent+teleport@automify.co.uk --cluster-name=tele.automify.co.uk | sudo tee /etc/teleport.yaml > /dev/null
Could not check the contents of /var/lib/teleport: open /var/lib/teleport: permission denied
The data directory may contain existing cluster state.
```

VM command history:
```
ubuntu@ip-10-0-1-49:~$ history
    1  exit
    2  sudo ufw allow 443/tcp
    3  exit
    4  sudo curl https://deb.releases.teleport.dev/teleport-pubkey.asc   -o /usr/share/keyrings/teleport-archive-keyring.asc
    5  echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] https://deb.releases.teleport.dev/ stable main" | sudo tee /etc/apt/sources.list.d/teleport.list > /dev/null
    6  sudo apt-get update
    7  sudo apt-get install teleport
    8  teleport configure --acme --acme-email=vincent+teleport@automify.co.uk --cluster-name=tele.automify.co.uk | sudo tee /etc/teleport.yaml > /dev/null
```

**Comment by vincentracine** [comment, marked as answer]
Running `sudo chmod 777 -R /var/lib/teleport` does fix this issue.

**Reply by webvictim** [reply]
A better fix would just be to add `sudo` to your `teleport configure` command. Teleport generally runs as `root` so will be able to read the directory itself when it runs, but the `ubuntu` user will not.

Leaving `/var/lib/teleport` world readable (777) isn't recommended, as any user on your system will be able to access Teleport's files/data.

**Reply by vincentracine** [reply]
Nice makes sense. I had read https://goteleport.com/docs/getting-started/linux-server/#before-you-begin-read-security-tips and it stated `Avoid using sudo in production environments unless it's necessary` so was trying to avoid it. I will use it to configure.
